trail/query/rest_views.py

```diff
@@ -28,35 +49,118 @@ def get_shared_non_query_keys(self, queryDict, *args, **kwargs):
 
         return queryDict, serializerParams, querysetParams
 
-    def get_non_query_keys(self, queryDict, *args, **kwargs):
+    def get_non_query_keys(self, queryDict):
+        """
+        Remove the keys and values not directly related to the queryDict.
+        and copies them into a queryset related parameter
+        and serializer related parameter dictionaries.
```

Comment:
Should probably specify that it's a no-op by default and that it allows for individual views to override this method to implement specialized behaviour based on keys found in the request.